lit.ts

```diff
@@ -478,6 +491,173 @@ export async function getPubkeyForAuthMethod({
+	const signer = getSigner();
+
+	const tx = await signer.sendTransaction({
+		to: recipientPublicKey,
+		value: parseEther(amount),
+	});
+
+	const reciept = await tx.wait();
+
+	console.log("Sent LIT tokens", reciept.blockHash);
+
+	return reciept.blockHash;
+}
+
+export async function mintCapacityCredits({
+	signer,
+}: {
+	signer: ethers.Wallet;
+}) {
+	if (config.network !== "habanero" && config.network !== "manzano") {
+		throw new Error(`Capacity credits not available on ${config.network}`);
+	}
+
+	const contract = await getContractFromWorker(config.network, 'RateLimitNFT', signer);
+
+	if (!contract) {
+		throw new Error('Contract is not available');
+	}
+
+	// set the expiration to midnight, 15 days from now
+	const timestamp = (Date.now() / 1000) + 15 * 24 * 60 * 60;
+	const futureDate = new Date(timestamp * 1000);
+	futureDate.setUTCHours(0, 0, 0, 0);
+
+	// Get the Unix timestamp in seconds
+	const expires = Math.floor(futureDate.getTime() / 1000);
+
+	const requestsPerKilosecond = 150;
+
+	let cost = parseEther('0.001');
+
+	try {
+		cost = await contract.functions.calculateCost(requestsPerKilosecond, expires);
+	} catch (e) {
+		console.log('Unable to estimate gas cost for minting capacity credits, using fallback');
+	}
+
+	const tx = await contract.functions.mint(expires, { value: cost.toString() });
+	const res = await tx.wait();
+
+	const tokenIdFromEvent = res.events[0].topics[1];
+
+	return { tx, capacityTokenId: tokenIdFromEvent };
+}
+
+function normalizeTokenURI(tokenURI: string) {
+	const base64 = tokenURI[0];
+
+	const data = base64.split('data:application/json;base64,')[1];
+	const dataToString = Buffer.from(data, 'base64').toString('binary');
+
+	return JSON.parse(dataToString);
+}
+
+function normalizeCapacity(capacity: any) {
+	const [requestsPerMillisecond, expiresAt] = capacity[0];
+
+	return {
+		requestsPerMillisecond: parseInt(requestsPerMillisecond.toString()),
+		expiresAt: {
+			timestamp: parseInt(expiresAt.toString()),
+		},
+	};
+}
+
+async function queryCapacityCredit(contract: ethers.Contract, tokenId: number) {
+	const [URI, capacity, isExpired] = await Promise.all([
+		contract.functions.tokenURI(tokenId).then(normalizeTokenURI),
+		contract.functions.capacity(tokenId).then(normalizeCapacity),
+		contract.functions.isExpired(tokenId),
+	]);
+
+	return {
+		tokenId,
+		URI,
+		capacity,
+		isExpired,
+	} as CapacityToken;
+}
+
+export async function queryCapacityCredits(signer: ethers.Wallet) {
+	if (config.network === "serrano" || config.network == "cayenne") {
+		throw new Error("Capacity credits are not available on Serrano");
+	}
+
+	const contract = await getContractFromWorker(config.network, 'RateLimitNFT');
+	const count = parseInt(await contract.functions.balanceOf(signer.address));
+
+	return await Promise.all([...new Array(count)].map(async (_, i) => {
+		return queryCapacityCredit(contract, i);
+	})) as CapacityToken[];
```

Comment:
can we add better error handling here? if this fails the stack trace is going to be hard to understand (and have unhandled promise rejection errors).

Reply:
There wouldn't be a stack trace, as the top-level catch in the API handler would capture the error, but I cleaned up the promise handling a bit so a theoretical stack trace would look like this:

```
/Users/jtary/Projects/lit/relay-server/lit.ts:589
		throw new Error(`Failed to fetch details for capacity token ${tokenId}: ${e}`);
        ^
Error: Failed to fetch details for capacity token 0: Test failure
    at queryCapacityCredit (/Users/jtary/Projects/lit/relay-server/lit.ts:589:9)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 0)
    at async addPaymentDelegationPayee (/Users/jtary/Projects/lit/relay-server/lit.ts:624:42)
    at async addPayeeHandler (/Users/jtary/Projects/lit/relay-server/routes/delegate/user.ts:24:16)
 ```